server/controllers/subject.js

```diff
@@ -0,0 +1,121 @@
+/* eslint-disable consistent-return */
```

Comment:
go through each and every one of these function and make each one an async/await function, then use the try/catch pattern to catch your errors.